homeassistant/components/device_tracker/bluetooth_tracker.py

```diff
@@ -64,27 +71,33 @@ def discover_devices():
     if track_new:
         for dev in discover_devices():
             if dev[0] not in devs_to_track and \
-               dev[0] not in devs_donot_track:
+                    dev[0] not in devs_donot_track:
                 devs_to_track.append(dev[0])
                 see_device(dev)
 
     interval = config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
 
+    request_rssi = config.get(CONF_REQUEST_RSSI, False)
+
     def update_bluetooth(now):
         """Lookup Bluetooth device and update status."""
         try:
             if track_new:
                 for dev in discover_devices():
                     if dev[0] not in devs_to_track and \
-                       dev[0] not in devs_donot_track:
+                            dev[0] not in devs_donot_track:
                         devs_to_track.append(dev[0])
             for mac in devs_to_track:
                 _LOGGER.debug("Scanning %s", mac)
-                result = bluetooth.lookup_name(mac, timeout=5)
-                if not result:
+                result = (bluetooth.lookup_name(mac, timeout=5),)
+                if request_rssi:
+                    rssi = BluetoothRSSI(mac).request_rssi()
+                    if rssi is not None:
+                        result = result + rssi
+                if result[0] is None:
                     # Could not lookup device name
                     continue
-                see_device((mac, result))
+                see_device((mac,) + result)
```

Comment:
This is also not easily comprehensible, what is the expected value for `see_device`? A tuple of mac + name? Maybe simply `see_device((mac, result))`?